runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -97,14 +97,13 @@ public void gc(long trimMark) {
      */
     public List<SMREntry> current() {
         ILogData data = streamView.current();
-        if (data != null) {
-            if (data.getType() == DataType.DATA
-                    && data.getPayload(runtime) instanceof ISMRConsumable) {
-                return ((ISMRConsumable) data.getPayload(runtime))
-                        .getSMRUpdates(streamView.getId());
-            }
+        if (data == null
+                || data.getType() != DataType.DATA
+                || !(data.getPayload(runtime) instanceof ISMRConsumable)) {
+            return new ArrayList<>();
```

Comment:
Can you use `Collections.emptyList()` instead?

Reply:
done